src/common/transformations/tests/common_optimizations/convert_u4_weights_zero_point_to_scalar.cpp

```diff
@@ -0,0 +1,202 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/convert_u4_weights_zero_point_to_scalar.hpp"
+
+#include <gtest/gtest.h>
+
+#include "common_test_utils/ov_test_utils.hpp"
+#include "openvino/core/model.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/pass/manager.hpp"
+
+using namespace testing;
+using namespace ov;
+
+TEST_F(TransformationTestsF, ConvertU4WeightsFloatZeroPointToScalar) {
+    auto weights_precision = ov::element::u4;
+    auto decompression_precision = ov::element::f32;
+    ov::Shape weights_shape{32, 128, 64};
+    ov::Shape decompression_shape{32, 1, 64};
+    {
+        auto weights = ov::op::v0::Constant::create(weights_precision, weights_shape, {4});
+        auto convert = std::make_shared<ov::op::v0::Convert>(weights, decompression_precision);
+        auto zero_point = ov::op::v0::Constant::create(decompression_precision, decompression_shape, {8});
```

Comment:
Do we need to cover cases with the data of zero_point being initialized by different values?

Reply:
This case is already [covered](https://github.com/openvinotoolkit/openvino/pull/20607/files/2464497d143796eae4f6140383cadc84a0e3d0de#diff-79b73c633f0b74da5c6d5d22cf05409c2d8c8941cdeb3a625eb900e342b994f2R103) by tests. In this case the transformation doesn't transform zero point